pkg/virt-operator/kubevirt.go

```diff
@@ -465,53 +462,87 @@ func (c *KubeVirtController) syncDeployment(kv *v1.KubeVirt) error {
 		logger.Errorf("Failed to create all resources: %v", err)
 		return err
 	}
+	util.RemoveCondition(kv, v1.KubeVirtConditionSynchronized)
 
 	if objectsAdded == 0 {
-		// deployment successful
-		kv.Status.Phase = v1.KubeVirtPhaseDeployed
 
-		// remove old conditions, add Created condition
-		kv.Status.Conditions = []v1.KubeVirtCondition{}
+		// add Created condition
 		util.UpdateCondition(kv, v1.KubeVirtConditionCreated, k8sv1.ConditionTrue, ConditionReasonDeploymentCreated, "All resources were created.")
-
 		logger.Info("All KubeVirt resources created")
-		return nil
+
+		// check if components are ready
+		if c.isReady() {
+			logger.Info("All KubeVirt components ready")
+			kv.Status.Phase = v1.KubeVirtPhaseDeployed
+			util.UpdateCondition(kv, v1.KubeVirtConditionReady, k8sv1.ConditionTrue, ConditionReasonDeploymentReady, "All components are ready.")
+			return nil
+		}
+		util.RemoveCondition(kv, v1.KubeVirtConditionReady)
+
+	} else {
+		util.RemoveCondition(kv, v1.KubeVirtConditionCreated)
 	}
 
 	logger.Info("Processed deployment for this round")
 	return nil
 }
 
+func (c *KubeVirtController) isReady() bool {
+
+	for _, obj := range c.stores.DeploymentCache.List() {
+		if deployment, ok := obj.(*appsv1.Deployment); ok {
+			var specReplicas int32 = 1
+			if deployment.Spec.Replicas != nil {
+				specReplicas = *deployment.Spec.Replicas
+			}
+			if specReplicas == deployment.Status.Replicas &&
```

Comment:
this doesn't look quite right. should this be 

```
if specReplicas != deployment.Status.Replicas || deployment.Status.Replicas != deployment.Status.ReadyReplicas
```
?

Reply:
oh yes, sure